iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/optional.inl

```diff
@@ -260,26 +271,36 @@ template <typename T>
 inline T& optional<T>::value() & noexcept
 {
     Expects(has_value());
-    return *static_cast<T*>(static_cast<void*>(&m_data));
+    // AXIVION Next Construct AutosarC++19_03-M5.2.8 : The optional has the type T defined
+    // during compile time and the type is unchangeable during the lifetime of the object.
+    // All accesses to the underlying data is done via the same static type and therefore the
+    // casts are always valid
+    return *static_cast<T*>(static_cast<void*>(m_data));
```

Comment:
We need to use `&m_data` here (for portability reasons).

Maybe it is necessary to use a `reinterpret_cast` (then one should be used).